src/routes/account/+page.server.ts

```diff
@@ -0,0 +1,107 @@
+import type { Actions, PageServerLoad } from './$types';
+import { error, redirect } from '@sveltejs/kit';
+import {
+  getStripeCustomerWithSubscriptions,
+  getBlendProSubscription,
+  getCustomerPortalSession,
+  createStripeSession,
+  isSubscribedToBlendPro,
+  isOrganizationMember,
+} from '$lib/server/subscriptionUtils';
+import { checkSessionAuth, getUserData, getUserOrganizations, isUserGlobalAdmin, readPath, writePath } from '$lib/server/firebaseUtils';
+
+let email = '';
+let name = '';
+let uid = '';
+
+export const load = (async ({ url, cookies }) => {
+  const actionParam = url.searchParams.get('action') ?? '';
+  uid = (await checkSessionAuth(cookies, {
+    loginRedirect: `account?${url.searchParams}`,
+    // authFunction: ({ uid: tokenUid }) => tokenUid === uid,
+  })).uid;
+
+  const userData = await getUserData(uid);
+  email = userData.email!;
+  name = userData.displayName ?? 'Blend User';
+  const isGlobalAdmin = isUserGlobalAdmin(uid); 
+  const organizations = getUserOrganizations(uid).then((orgIds) => {
+    return Promise.all(
+      (orgIds || []).map(async (orgId: string) => {
+        const organization = await readPath<Database.Organization>(`/organizations/${orgId}`);
+        return {  
+          id: orgId,
+          name: organization?.public.name,
+          role: (await isGlobalAdmin) ? 'admin' : organization?.private?.members?.[uid]?.role,
+        };
+      }),
+    );
+  });
+  const customer = await getStripeCustomerWithSubscriptions(uid);
+
+  // Redirect to Stripe Checkout if necessary
+  if(!isSubscribedToBlendPro(customer) && !(await isOrganizationMember(uid))) {
+    switch (actionParam) {
+      case 'upgrade': {
+        const stripeSession = await createStripeSession(uid, email, name, url.origin);
+        throw redirect(303, stripeSession.url!);
+      }
+      case 'choosePlan':
+        throw redirect(303, '/account/plan');
+      default: {
+        break;
+      }
+    }
+  }
```

Comment:
Can simplify slightly (idk if the linter required a default, if so disregard that edit) and skip this when it is not present, which we should in order to skip the network requests. 
```suggestion
  if(actionParam && !isSubscribedToBlendPro(customer) && !(await isOrganizationMember(uid))) {
    switch (actionParam) {
      case 'upgrade': {
        const stripeSession = await createStripeSession(uid, email, name, url.origin);
        throw redirect(303, stripeSession.url!);
      }
      case 'choosePlan':
        throw redirect(303, '/account/plan');
    }
  }
```